manager/controllers/app/plotter_generator.go

```diff
@@ -201,6 +204,55 @@ func (p *PlotterGenerator) addStep(element *ResolvedEdge, datasetID string, api
 	return steps
 }
 
+// Create a new asset and register it to the catalog. Used when the IsNewDataSet flag is true
+func (p *PlotterGenerator) HandleNewAsset(item *DataInfo, application *app.FybrikApplication, selection *Solution,
+	plotterSpec *app.PlotterSpec) error {
+	var err error
+	if item.DataDetails.Details.DataFormat != "" {
+		return nil
+	}
+	var sinkDataStore *app.DataStore
+	selectionLen := len(selection.DataPath)
+	element := selection.DataPath[selectionLen-1]
```

Comment:
It would be cleaner to go over the DataPath and look for an element with a non-empty storage account instead of relying on it to be the last.

Reply:
Done. Thanks